service-base/src/main/java/org/eclipse/hono/service/auth/X509CertificateChainValidator.java

```diff
@@ -39,4 +41,44 @@
      * @throws IllegalArgumentException if the chain is empty.
      */
     Future<Void> validate(List<X509Certificate> chain, TrustAnchor trustAnchor);
+
+    /**
+     * Validates a certificate path based on a list of trust anchors.
+     * <p>
+     * This default implementation simply calls {@link #validate(List, TrustAnchor)} for each trust anchor in the list.
+     * 
+     * @param chain The certificate chain to validate. The end certificate
+     *              must be at position 0.
+     * @param trustAnchors The list of trust anchors to use for validating the chain.
+     * @return A completed future if the certificate path is valid for any of the trust anchors.
+     *         Otherwise, the future will be failed when all the trust anchors cannot be validated by the certificate path.
+     *
+     * @throws NullPointerException if any of the parameters are {@code null}.
+     * @throws IllegalArgumentException if the chain and/or trust anchors is empty.
+     */
+    default Future<Void> validate(List<X509Certificate> chain, List<TrustAnchor> trustAnchors) {
```

Comment:
I do not think that we need to provide a default implementation here. Just defining the method will do. In fact, most implementations will probably delegate to this variant as indicated earlier by you, instead of the other way around ...

Reply:
> I do not think that we need to provide a default implementation here. Just defining the method will do.

ok and this will mean that the `X509CertificateChainValidator` can no longer be a functional interface -> i updated the patch